src/qpx_mmc_css.cpp

```diff
@@ -0,0 +1,1909 @@
+      k[4] = B[4] ^ tmp;
+      nPossibleK1 = K1table[K1TABLEWIDTH * (256 * B[0] + C[1])];
+
+      /* Try out all possible values for k[1] */
+      for (nTry = 0; nTry < nPossibleK1; nTry++) {
+        k[1] = K1table[K1TABLEWIDTH * (256 * B[0] + C[1]) + nTry + 1];
+        B[1] = tmp5 ^ k[1];
+
+        /* reconstruct output from LFSR2 */
+        tmp3 = (0x100 + k[0] - out1[0]);
+        out2[0] = tmp3 & 0xff;
+        tmp3 = tmp3 & 0x100 ? 0x100 : 0xff;
+        tmp3 = (tmp3 + k[1] - out1[1]);
+        out2[1] = tmp3 & 0xff;
+        tmp3 = (0x100 + k[4] - out1[4]);
+        out2[4] = tmp3 & 0xff; /* Can be 1 off  */
+
+        /* test first possible out2[4] */
+        tmp4 = (out2[0] << 16) | (out2[1] << 8) | out2[4];
+        tmp4 = BigTable[tmp4];
+        C[2] = tmp4 & 0xff;
+        C[3] = (tmp4 >> 8) & 0xff;
+        C[4] = (tmp4 >> 16) & 0xff;
+        B[3] = p_css_tab1[B[4]] ^ k[4] ^ C[4];
+        k[3] = p_disc_key[2] ^ p_css_tab1[p_disc_key[3]] ^ B[3];
+        B[2] = p_css_tab1[B[3]] ^ k[3] ^ C[3];
+        k[2] = p_disc_key[1] ^ p_css_tab1[p_disc_key[2]] ^ B[2];
+
+        if ((B[1] ^ p_css_tab1[B[2]] ^ k[2]) == C[2]) {
+          if (!css_investigate(&p_disc_key[0], &C[0])) {
+            goto end;
+          }
+        }
+
+        /* Test second possible out2[4] */
+        out2[4] = (out2[4] + 0xff) & 0xff;
+        tmp4 = (out2[0] << 16) | (out2[1] << 8) | out2[4];
+        tmp4 = BigTable[tmp4];
+        C[2] = tmp4 & 0xff;
+        C[3] = (tmp4 >> 8) & 0xff;
+        C[4] = (tmp4 >> 16) & 0xff;
+        B[3] = p_css_tab1[B[4]] ^ k[4] ^ C[4];
+        k[3] = p_disc_key[2] ^ p_css_tab1[p_disc_key[3]] ^ B[3];
+        B[2] = p_css_tab1[B[3]] ^ k[3] ^ C[3];
+        k[2] = p_disc_key[1] ^ p_css_tab1[p_disc_key[2]] ^ B[2];
+
+        if ((B[1] ^ p_css_tab1[B[2]] ^ k[2]) == C[2]) {
+          if (!css_investigate(&p_disc_key[0], &C[0])) {
+            goto end;
+          }
+        }
+      }
+    }
+  }
+
+end:
+
+  memcpy(p_disc_key, &C[0], DVD_KEY_SIZE);
+
+  free(K1table);
+  free(BigTable);
+
+  return 0;
+}
+
+/*****************************************************************************
+ * RecoverTitleKey: (title) key recovery from cipher and plain text
+ * Function designed by Frank Stevenson
+ *****************************************************************************
+ * Called from Attack* which are in turn called by CrackTitleKey.  Given
+ * a guessed(?) plain text and the cipher text.  Returns -1 on failure.
+ *****************************************************************************/
+static int css_RecoverTitleKey(int i_start, unsigned char const *p_crypted,
+                               unsigned char const *p_decrypted,
+                               unsigned char const *p_sector_seed,
+                               unsigned char *p_key) {
+  unsigned char p_buffer[10];
+  unsigned int i_t1, i_t2, i_t3, i_t4, i_t5, i_t6;
+  unsigned int i_try;
+  unsigned int i_candidate;
```

Comment:
Codacy found an issue: [The scope of the variable 'i_candidate' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)